pkg/constants/constants.go

```diff
@@ -58,6 +58,9 @@ const (
 
 	// GCPCredentialsName is the name of the GCP credentials file or secret key.
 	GCPCredentialsName = "osServiceAccount.json"
+
+	// OwnerLabel is the label that is used to identify the owner of a runtime object.
+	OwnerLabel = "hive.openshift.io/owner"
```

Comment:
I don't think that you want to use a generic label. This is going to conflict for the types that may be owned by multiple different types of owners. For example, jobs can be owned by either a clusterdeployment or a clusterprovision. What you should be looking for is a specific label placed on a resource that is doing a specific task. For example, to re-establish the owner reference for the imageset job, you should be looking for a job with a labels "hive.openshift.io/imageset=true" and "hive.openshift.io/cluster-deployment-name=CD_NAME", where CD_NAME is the name of the clusterdeployment being reconciled.